File: 0006-zigzag-conversion/0006-zigzag-conversion.cpp
class Solution {
public:
    string convert(string s, int numRows) {
        int size = s.size() ;

        if(numRows == size || numRows == 1) return s;

        int row = 0 , index = -1;

        vector<string>temp(numRows , "");

        for(char ch : s){
            temp[row].push_back(ch);


            if(row == 0 || row == numRows-1){
                index *= -1;
            }
            row += index;
        }


        string res = "";

        for(string str : temp){
            res += str;
        }

        return res;
    }
};

File: 0006-zigzag-conversion/0006-zigzag-conversion.py
class Solution(object):
    def convert(self, s, numRows):
        """
        :type s: str
        :type numRows: int
        :rtype: str
        """
        size = len(s)

        if size <= numRows or numRows == 1:
            return s

        map = [""] * numRows

        index = 0
        direction = 1 

        for char in s:
            map[index] += char
            if index == 0:
                direction = 1
            elif index == numRows - 1:
                direction = -1
            index += direction

        return "".join(map)      